tests/components/light/test_device_trigger.py

```diff
@@ -212,15 +212,12 @@ async def test_if_fires_on_state_change(
                     "action": {
                         "service": "test.automation",
                         "data_template": {
-                            "some": "turn_on {{ trigger.%s }}"
-                            % "}} - {{ trigger.".join(
-                                (
-                                    "platform",
-                                    "entity_id",
-                                    "from_state.state",
-                                    "to_state.state",
-                                    "for",
-                                )
+                            "some": (
+                                "turn_on {{ trigger.platform }}"
+                                " - {{ trigger.entity_id }}"
+                                " - {{ trigger.from_state.state }}"
+                                " - {{ trigger.to_state.state }}"
+                                " - {{ trigger.for }}"
```

Comment:
May be a list assignment and join would be better here?

Reply:
What do you mean exactly? We have this style kind of in the scaffold templates as well, and the format+join combo is a lot harder to read.

Reply:
Yes, but there is quite some instances using the same list of attributes.

Reply:
I committed an example, did you mean this?

Reply:
No, I meant just to put the list in a constant and use join, the list seems to be reused:
```
[
    "platform",
    "entity_id",
    "from_state.state",
    "to_state.state",
    "for",
]
```

Reply:
So what I am aiming to is to keep using `join` in these cases, but with the const.

Reply:
But then we still use join, which this rule wants to remove.

Reply:
But what is wrong with that, if the list contants are repeated then using a join will be the better choice

Reply:
It can be hard to read, especially when combined with format-strings like here.

Reply:
That is the reason why to kae out the list of constants and reuse them. That will also improve readability imo

Reply:
I will no longer block this PR though, as it seems just a different opinion about a small thing.